modules/tracking/include/opencv2/tracking/kalman_filters.hpp

```diff
@@ -0,0 +1,215 @@
+
+    /** The function performs prediction step of the algorithm
+    * @param control - the current control vector,
+    * @return the predicted estimate of the state.
+    */
+    CV_WRAP virtual Mat predict( const Mat& control = Mat() ) = 0;
+
+    /** The function performs correction step of the algorithm
+    * @param measurement - the current measurement vector,
+    * @return the corrected estimate of the state.
+    */
+    CV_WRAP virtual Mat correct( const Mat& measurement ) = 0;
+
+    /**
+    * @return the process noise cross-covariance matrix.
+    */
+    CV_WRAP virtual Mat getProcessNoiseCov() const = 0;
+
+    /**
+    * @return the measurement noise cross-covariance matrix.
+    */
+    CV_WRAP virtual Mat getMeasurementNoiseCov() const = 0;
+
+    /**
+    * @return the current estimate of the state.
+    */
+    CV_WRAP virtual Mat getState() const = 0;
+};
+
+
+/** The function for computing the next state from the previous state
+* @param x_k - previous state vector,
+* @param u_k - control vector,
+* @param v_k - noise vector,
+* @param x_kplus1 - next state vector.
+*/
+typedef void( * UKFStateFunction )( const Mat& x_k, const Mat& u_k, const Mat& v_k, Mat& x_kplus1 );
+
+/** The function for computing the measurement from the state
+* @param x_k - state vector,
+* @param n_k - noise vector,
+* @param z_k - measurement vector.
+*/
+typedef void( * UKFMeasFunction )( const Mat& x_k, const Mat& n_k, Mat& z_k );
+
+
+/** @brief Unscented Kalman filter parameters.
+* The class for initialization parameters of Unscented Kalman filter
+*/
+class CV_EXPORTS_W UnscentedKalmanFilterParams
+{
+public:
+
+    CV_PROP_RW int DP;                                     //!< Dimensionality of the state vector.
+    CV_PROP_RW int MP;                                     //!< Dimensionality of the measurement vector.
+    CV_PROP_RW int CP;                                     //!< Dimensionality of the control vector.
+    CV_PROP_RW int dataType;                               //!< Type of elements of vectors and matrices, default is CV_64F.
+
+    CV_PROP_RW Mat stateInit;                              //!< Initial state, DP x 1, default is zero.
+    CV_PROP_RW Mat errorCovInit;                           //!< State estimate cross-covariance matrix, DP x DP, default is identity.
+
+    CV_PROP_RW Mat processNoiseCov;                        //!< Process noise cross-covariance matrix, DP x DP.
+    CV_PROP_RW Mat measurementNoiseCov;                    //!< Measurement noise cross-covariance matrix, MP x MP.
+
+    // Parameters of algorithm
+    CV_PROP_RW double alpha;                               //!< Default is 1e-3.
+    CV_PROP_RW double k;                                   //!< Default is 0.
+    CV_PROP_RW double beta;                                //!< Default is 2.0.
+
+    //Functions
+    CV_PROP_RW UKFStateFunction f;                            //!< Function for computing the next state from the previous state.
```

Comment:
there is no way to wrap function pointers in Python, Java etc., so please, do not put CV_PROP_RW in front of such members